collect_app/src/test/java/org/odk/collect/android/projects/ProjectDeleterTest.kt

```diff
@@ -1,32 +1,114 @@
 package org.odk.collect.android.projects
 
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.instanceOf
+import org.hamcrest.Matchers.nullValue
 import org.junit.Test
 import org.mockito.kotlin.doReturn
 import org.mockito.kotlin.mock
 import org.mockito.kotlin.verify
 import org.odk.collect.android.backgroundwork.FormUpdateScheduler
 import org.odk.collect.android.backgroundwork.InstanceSubmitScheduler
+import org.odk.collect.android.database.instances.DatabaseInstancesRepository
+import org.odk.collect.forms.instances.Instance
 import org.odk.collect.projects.Project
+import org.odk.collect.projects.ProjectsRepository
 
 class ProjectDeleterTest {
+    private val project1 = Project.Saved("1", "1", "1", "#ffffff")
+    private val project2 = Project.Saved("2", "2", "2", "#cccccc")
+    private val currentProjectProvider = mock<CurrentProjectProvider> {
+        on { getCurrentProject() } doReturn project1
+    }
+    private val instancesRepository = mock<DatabaseInstancesRepository> {
+        on { allUnsent } doReturn emptyList()
```

Comment:
I think it might be nicer to just use `InMemInstancesRepository` as it means there is less enforcing of the implementation (we just set up the state rather than enforce what method is called). What do you think?

Reply:
Yeas, done.